scripts/mbedtls_framework/c_parsing_helper.py

```diff
@@ -60,18 +62,37 @@ def __init__(self, #pylint: disable=too-many-arguments
                  qualifiers: Iterable[str],
                  return_type: str,
                  name: str,
-                 arguments: List[str]) -> None:
+                 arguments: List[str],
+                 doc: str = "") -> None:
+
         self.filename = filename
         self.line_number = line_number
         self.qualifiers = frozenset(qualifiers)
         self.return_type = return_type
         self.name = name
         self.arguments = [ArgumentInfo(arg) for arg in arguments]
+        self.doc = doc
 
     def returns_void(self) -> bool:
         """Whether the function returns void."""
         return bool(self.VOID_RE.search(self.return_type))
 
+    def __str__(self) -> str:
+        _str_args = [str(a) for a in self.arguments]
+        _str = "{} {} {}({})".format(" ".join(self.qualifiers),
```

Comment:
Minor: This should be called `str_`, per [PEP 8](https://peps.python.org/pep-0008/#descriptive-naming-styles), if the intent is to call it `str` without shadowing the built-in function by that name.

Reply:
I actually want to override the built-in function. It serves no use as is, but its helpfull for print debugging or for writing onto a c file if we override the `__str__` representation.

Reply:
One of us is not understanding the other, I'm not sure which.

I know that the `__str__` method customizes the behavior of `str` on objects of this class. But it's irrelevant to my comment.

My comment is about the local variable `_str`. The widely adopted conventions recorded in PEP 8 state that local variables should not begin with an underscore, except to indicate that they are possibly uninitialized (which stops linters such as pylint warning about that). So the variable should not be called `_str`.

It appears that a natural name for this variable would be `str`. However, `str` is not a good variable name for a different reason: this is a built-in function, and while it's possible to override built-in functions, it can have very surprising effects if you accidentally both override a built-in function and attempt to use it. Hence PEP 8 recommends to not override the names of built-in functions, and if you want that name, to **append** (not prepend) an underscore. Thus the variable should be called `str_` (or something else altogether like `string`).